http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala

```diff
@@ -500,6 +493,41 @@ final class LeoAppServiceInterp[F[_]: Parallel](
     )
   }
 
+  private[service] def validateGalaxy(googleProject: GoogleProject,
+                                      diskSize: Option[DiskSize],
+                                      machineTypeName: MachineTypeName)(
+    implicit as: Ask[F, AppContext]
+  ): F[AppMachineType] =
+    for {
+      ctx <- as.ask
+      _ <- diskSize
+        .traverse_ { size =>
+          if (size.gb < leoKubernetesConfig.galaxyDiskConfig.nfsMinimumDiskSizeGB.gb) {
+            F.raiseError[Unit](
+              BadRequestException(
+                s"Galaxy disk must be at least ${leoKubernetesConfig.galaxyDiskConfig.nfsMinimumDiskSizeGB}",
+                Some(ctx.traceId)
+              )
+            )
+          } else F.unit
+        }
+      machineType <- computeService
+        .getMachineType(googleProject,
+                        ZoneName("us-central1-a"),
+                        machineTypeName) //TODO: if use non `us-central1-a` zone for galaxy, this needs to be udpated
+        .flatMap(opt =>
+          F.fromOption(opt,
+                       new LeoException(s"can't find machine config for ${ctx.requestUri}",
```

Comment:
Maybe this should be `BadRequestException("Unknown machine type ${machineTypeName.asString}")`